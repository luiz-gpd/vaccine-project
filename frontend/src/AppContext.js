import { createContext } from 'react'

const AppContext = createContext();


// const reducer = (state, action) => {
//     switch (action.type) {
//         case "EXEMPLE": {
//             return {
//                 ...state,
//                 loggedUser: action.payload,
//             };
//         }

//         default: {
//             return state;
//         }
//     }
// }

export default AppContext