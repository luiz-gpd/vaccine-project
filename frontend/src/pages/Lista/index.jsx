import React, { useEffect, useState } from 'react'
import Page from '../../components/Page'
import api from '../../utils/api'
import ClayTable from '@clayui/table'
import ClayLoadingIndicator from '@clayui/loading-indicator'
import { ClayToggle } from '@clayui/form'
import { useModal } from '@clayui/modal'
import { ClayButtonWithIcon } from '@clayui/button'
import Modal from '../../components/Modal'
import moment from 'moment'

const Lista = () => {

    const initialState = {
        name:"",
        consultationDate:"",
        consultationTime:"",
        consultInfo:"",
    }

    const [users, setUsers] = useState([])
    const [loading, setLoading] = useState(false);
    const [visible, setVisible] = useState(false);  
    const [form, setForm] = useState(initialState);
    const { observer, onClose } = useModal({
        onClose: () => setVisible(false),
    });

    const getUsers = () => {
        setLoading(true);
        api.get('/user').then((response) => {
            const { data } = response;
            setUsers(data);
        }).catch((e) => {
            console.log(e);
        });
        setLoading(false);
    };

    useEffect(() => {
        getUsers();
    }, [])

    const onToggle = async (name, date, time) => {
        await api.put(`/user/${name}/${date}/${time}`);
        getUsers();
    }

    const getModalInfo = ( name, date, time, info ) => {
        setVisible(true);
        setForm({
            name:name,
            consultationDate:date,
            consultationTime:time,
            consultInfo:info
        })
    }

    return (
        <Page title="Lista de Agendamentos">
            {loading ? (
                <ClayLoadingIndicator />
            ) : (
                <>
                    <ClayTable>
                        <ClayTable.Head>
                            <ClayTable.Row>
                                <ClayTable.Cell headingCell>Nome</ClayTable.Cell>
                                <ClayTable.Cell headingCell>Idade</ClayTable.Cell>
                                <ClayTable.Cell headingCell>Data da Vacina</ClayTable.Cell>
                                <ClayTable.Cell headingCell>Horário da Vacina</ClayTable.Cell>
                                <ClayTable.Cell headingCell>Atendimento</ClayTable.Cell>
                            </ClayTable.Row>
                        </ClayTable.Head>
                        <ClayTable.Body>
                            {users.map((user, index) => (
                                <ClayTable.Row key={index}>
                                    <ClayTable.Cell>{user.name}</ClayTable.Cell>
                                    <ClayTable.Cell>{user.age}</ClayTable.Cell>
                                    <ClayTable.Cell>{moment(user.consultationDate).format("DD/MM/yyyy")}</ClayTable.Cell>
                                    <ClayTable.Cell>{user.consultationTime}</ClayTable.Cell>
                                    <ClayTable.Cell>
                                        <ClayToggle label={user.attended ? "Realizado" : "Não foi realizado"}
                                            // disabled={(user.consultationDate > new Date()) ? "" : "anything"}
                                            toggled={user.attended}
                                            onToggle={() => onToggle(user.name, user.consultationDate, user.consultationTime)} />
                                        {user.attended && <ClayButtonWithIcon className="btn btn-primary btn-sm ml-2"
                                        symbol="comments" onClick={() => getModalInfo(user.name, user.consultationDate, user.consultationTime, user.consultInfo)}/>}
                                    </ClayTable.Cell>
                                </ClayTable.Row>

                            ))}
                        </ClayTable.Body>
                    </ClayTable>
                    <Modal
                        visible={visible}
                        onClose={onClose}
                        observer={observer}
                        title="Dados da consulta"
                        name={form.name}
                        date={form.consultationDate}
                        time={form.consultationTime}
                        >{form.consultInfo}
            </Modal>
                </>)}
        </Page>
    )
}

export default Lista;
