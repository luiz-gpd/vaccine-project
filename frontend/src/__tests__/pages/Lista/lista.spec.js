import Lista from "../../../pages/Lista"

import AppContext, { initialState } from "../../../AppContext"

import { render, fireEvent, waitFor } from "@testing-library/react"
import { BrowserRouter } from "react-router-dom";

const ListaWrapper = ({ state = initialState, dispatch }) => (
    <AppContext.Provider value={[state, dispatch]}>
        <BrowserRouter>
            <Lista />
        </BrowserRouter>
    </AppContext.Provider>
);

describe("Lista page", () => {
    it("renders", () => {
        const { asFragment } = render(<ListaWrapper />);

        expect(asFragment()).toMatchSnapshot();
    });

    it("Button should submit when clicked", async () => {
        const dispatch = jest.fn();

        const { debug } = render(
            <ListaWrapper
                state={{
                    ...initialState,
                    search: "search",
                    modalInfo: {
                        userId: "6073716197bf474d08d1098f",
                        info: "Tudo bem",
                        visible: false
                    },
                }}
                users={ {
                    name:"Luiz",
                    age:"19"
                } }
                dispatch={dispatch}
            />
        );
        // const fieldNode = await waitFor(
        //     () => queryByTestId('search-input')
        // )
        // fireEvent.change(
        //     fieldNode,
        //     { target: { value: 'test' } }
        // )

        // expect(fieldNode.value).toEqual(test)

        debug(undefined, 300000);

    });
});