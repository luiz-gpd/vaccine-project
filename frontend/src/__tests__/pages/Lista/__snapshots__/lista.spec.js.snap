// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lista page renders 1`] = `
<DocumentFragment>
  <div
    class="mt-4 container"
  >
    <div
      class="row justify-content-between"
    >
      <div
        class="col"
      >
        <h1>
          Lista de Agendamentos
        </h1>
      </div>
      <div
        class="col-3"
      >
        <img
          alt="Logo do site"
          src="VaciCuidar_logo.png"
          width="250px"
        />
      </div>
    </div>
    <span>
       Não há nenhuma consulta agendada
    </span>
  </div>
</DocumentFragment>
`;
